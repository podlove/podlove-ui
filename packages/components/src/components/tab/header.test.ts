import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import TabHeader from './Header.vue';

describe('TabHeader', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        }
      });

      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        }
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    test(`should render with prop backgroundActive`, async () => {
      const wrapper = await mount(TabHeader, {
        slots: {
          default: '<span>Inner Tab Header</span>'
        },
        props: {
          backgroundActive: '#fff'
        }
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
