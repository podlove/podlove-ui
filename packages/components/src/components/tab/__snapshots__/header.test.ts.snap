// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabHeader > props > should render with prop backgroundActive 1`] = `
<ul
  backgroundactive="#fff"
  class="podlove-component-tab-header relative w-full flex items-center justify-center font-hairline h-8"
  role="tablist"
>
  <span
    class="block absolute pointer-events-none bottom-0 left-0 right-0 h-4 z-30 -mb-4"
  />
  
  <span>
    Inner Tab Header
  </span>
  
</ul>
`;

exports[`TabHeader > props > should render without props 1`] = `
<ul
  class="podlove-component-tab-header relative w-full flex items-center justify-center font-hairline h-8"
  role="tablist"
>
  <span
    class="block absolute pointer-events-none bottom-0 left-0 right-0 h-4 z-30 -mb-4"
  />
  
  <span>
    Inner Tab Header
  </span>
  
</ul>
`;
